social-tw/social-tw-website
# [Frontend] Feature/ust 72 refactor voting logic

## Summary
This pull request refactors the voting logic for the frontend application. It extracts the handleVoteEvent function into a shared module and optimizes the Post component's voting handling, making the code cleaner and more maintainable.

## Linked Issue
This PR addresses the issue #306 

## Details
Extracted the handleVoteEvent function into a shared module (useVoteEvents.ts).
Optimized the Post component's voting logic to be more efficient and maintainable.
Ensured the postInfo block remains intact and utilized appropriately within the component.
Impacted Areas

## The following files or modules are affected by this PR:

src/pages/Home/HomePostList.tsx
src/components/post/Post.tsx
src/hooks/useVoteEvents.ts (newly added)

## Tests
Unit tests for the refactored handleVoteEvent function.
Integration tests for the Post component to ensure voting logic works as expected.
Manual testing to verify the UI changes and overall functionality.
Possible Impacts
This refactoring should not impact any existing features negatively. However, there might be potential backward compatibility issues if other components rely on the old voting logic.

## Verification Steps
Navigate to the home page and verify that posts are displayed correctly.
Perform upvote and downvote actions on various posts.
Ensure that the vote counts update correctly and persist across page reloads.
Verify that the extracted handleVoteEvent function works as expected by checking the vote state updates.

## Checklist
